surveyor/jetstream_advisory.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2020 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package surveyor
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/nats-io/jsm.go"
+	"github.com/nats-io/jsm.go/api"
+	"github.com/nats-io/nats.go"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type JSAdvisoryListener struct {
+	nc    *nats.Conn
+	opts  *jsAdvisoryOptions
+	sopts *Options
+}
+
+type jsAdvisoryOptions struct {
+	AccountName string `json:"name"`
+	Credentials string `json:"credentials"`
+}
+
+func (o *jsAdvisoryOptions) Validate() error {
+	errs := []string{}
+
+	if o.AccountName == "" {
```

Comment:
Allow $G and old style auth?

Reply:
user/pass is supported , the name here is just for users to know what they are looking at, something for them better than $G